### PR TITLE
build: improve update script error handling

### PR DESCRIPTION
**Pull request by MikeMcC399**
This PR improves the error handling for the update scripts called by

```bash
npm run update:cypress
```

It was previously possible for them to fail silently, particularly if `pnpm` or `yarn` were not installed.  This situation can occur if `nvm` is used to manage `Node.js` versions, since global installations of `pnpm` and `yarn` are tied to a specific `Node.js` version and not carried over, for instance if `Node.js` is updated to the latest LTS version.

The scripts now check that the following commands are available:

- npm
- pnpm
- yarn

and advise installation if they are missing.

## Verification

On Ubuntu and Windows 11 Pro.

1. With no `npm` available (for instance by uninstalling `nvm`). `npm -v` should produce "command not found", execute

```bash
./scripts/update-cypress-latest.sh
```

and confirm output of message to install `Node.js` or `nvm`.

2. Install [Node.js LTS](https://nodejs.org/en/) (currently 18.5.0) either directly or via `nvm`.

On Windows execute

```bash
npm config set script-shell "C:\\Program Files\\git\\bin\\bash.exe" --location user
```

Execute

```bash
npm ci
npm run update:cypress
```

Confirm that the script informs that `npm` is installed, and that `pnpm` and `yarn` are not installed.

3. Follow the instructions to install `pnpm` and execute the script again.

Confirm that the script informs that `npm` is installed, and that `yarn` is not installed.

4. Follow the instructions to install `yarn` and execute the script again.

This time the script should run to completion and update all examples to use `cypress@12.8.0` (excepting those in the [examples/v9](https://github.com/cypress-io/github-action/tree/master/examples/v9) directory).

### CODE REVIEW

File: scripts/check-package-manager-npm.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+# This script checks that the prerequisite
+# npm is installed
+
+if ! command -v npm &> /dev/null
+then
+    echo "**npm is required and not installed**"
+    echo "install Node.js LTS from:"
+    echo "https://nodejs.org/en/"
+    echo "or install and use nvm"
```
Comment (AtofStryker) [inline scripts/check-package-manager-npm.sh:10]: should we also link out to [nvm](https://github.com/nvm-sh/nvm) as well?

**Review by AtofStryker — APPROVED** [review]
LGTM. I wonder if the script could also check if `nvm` is installed and if not, install it and check if the system node is set to `lts`, otherwise bail

File: scripts/check-package-manager-npm.sh (hunk 1 above)
Comment (MikeMcC399) [reply scripts/check-package-manager-npm.sh:10]: @AtofStryker 

`nvm` is not a prerequisite, so I don't really want to start handling that. Also there are different versions, one for Linux and a different one for Windows.

There may also be cases where an end-user wants to use the script and does **not** want to use the LTS version.

So I think it is OK the way it is.

The CI [.github/workflows/example-node-versions.yml](https://github.com/cypress-io/github-action/blob/master/.github/workflows/example-node-versions.yml) runs tests on `Node.js` `14`, `16`, `18`, `19`, so I don't think that there is any significant exposure by not checking deeper at this point.

Comment (AtofStryker) [reply scripts/check-package-manager-npm.sh:10]: @MikeMcC399 I agree. I don't think it's necessary here and the jobs itself should mitigate anything if a wrong version were used. I'm not too worried about it 😄 .

Comment (MikeMcC399) [reply scripts/check-package-manager-npm.sh:10]: The main reason for this update was to help me from making avoidable mistakes using my own scripts! 😆

**Comment by AtofStryker** [comment]
@lmiller1990 you mind taking a look at this PR? the Node v19 job for Cypress 9 is failing since the actual `.gz` file 404s on Node's site. I'm not sure why this isn't failing for the Cypress@12 jobs though for v19.

**Comment by MikeMcC399** [comment]
@AtofStryker 

> the Node v19 job for Cypress 9 is failing since the actual `.gz` file 404s on Node's site. I'm not sure why this isn't failing for the Cypress@12 jobs though for v19.

The results of [.github/workflows/example-node-versions.yml](https://github.com/cypress-io/github-action/blob/master/.github/workflows/example-node-versions.yml) are flaky for multiple PRs.

In the run for this PR https://github.com/cypress-io/github-action/actions/runs/4429718318

https://github.com/cypress-io/github-action/actions/runs/4429718318/jobs/7770543697 fails with

```
Attempting to download 19...
Not found in manifest. Falling back to download directly from Node
Acquiring 19.[8](https://github.com/cypress-io/github-action/actions/runs/4429718318/jobs/7770543697#step:2:9).1 - x64 from https://nodejs.org/dist/v1[9](https://github.com/cypress-io/github-action/actions/runs/4429718318/jobs/7770543697#step:2:10).8.1/node-v19.8.1-linux-x64.tar.gz
Downloading only node binary from https://nodejs.org/dist/v19.8.1/win-x64/node.exe
Error: Unexpected HTTP response: 404
```

which doesn't quite make sense because this is an Ubuntu runner, not a `win-x64` one

---
https://github.com/cypress-io/github-action/actions/runs/4429718318/jobs/7770541047 succeeds, although there shouldn't be any difference between this run and the previous one which failed;

```
Attempting to download 19...
Not found in manifest. Falling back to download directly from Node
Acquiring 19.[8](https://github.com/cypress-io/github-action/actions/runs/4429718318/jobs/7770541047#step:2:9).1 - x64 from https://nodejs.org/dist/v1[9](https://github.com/cypress-io/github-action/actions/runs/4429718318/jobs/7770541047#step:2:10).8.1/node-v19.8.1-linux-x64.tar.gz
Extracting ...
/usr/bin/tar xz --strip 1 --warning=no-unknown-keyword -C /home/runner/work/_temp/fc199[11](https://github.com/cypress-io/github-action/actions/runs/4429718318/jobs/7770541047#step:2:12)2-ab1c-4734-a359-3f80dd99f0bc -f /home/runner/work/_temp/bcad[14](https://github.com/cypress-io/github-action/actions/runs/4429718318/jobs/7770541047#step:2:15)a7-7d05-4486-bc62-cb283a[17](https://github.com/cypress-io/github-action/actions/runs/4429718318/jobs/7770541047#step:2:18)27de
Adding to the cache ...
Done
Environment details
  node: v19.8.1
  npm: 9.5.1
  yarn: 1.22.19
```

I'm not sure if this is a GitHub issue or a Node.js issue at the moment. It could be a flaky node or load-balancer for instance.

If it's still happening tomorrow I would take the problem to GitHub / Node.js support to look at.

**Comment by AtofStryker** [comment]
> I'm not sure if this is a GitHub issue or a Node.js issue at the moment. It could be a flaky node or load-balancer for instance.
> 
> If it's still happening tomorrow I would take the problem to GitHub / Node.js support to look at.

I think its a Node.js issue. When I actually try to download the file itself from the browser it fails with a 404. Almost seems like a faulty load balancer or something of sorts. If it is passing for the Cypress latest v19 job, I would assume the load balancer. My guess is this should be fixed either by a node version bump or someone adjusting the load balancer in the near future.

I wouldn't worry too much about it, but as you said if it is still happening tomorrow it might be worth looking at, but I don't know how we would be able to fix that besides filing an issue.

**Review by lmiller1990 — APPROVED** [review]

**Comment by lmiller1990** [comment]
Seems like it's okay now.

**Comment by MikeMcC399** [comment]
@AtofStryker 
`curl -I https://nodejs.org/dist/v19.8.1/win-x64/node.exe` today produces an `HTTP/2 200` response, whereas yesterday it was responding with `HTTP/2 404`, so it seems that Node.js / cloudflare have fixed their issue.

**Pull request merged**